core/src/test/java/io/questdb/test/metrics/QueryTracingSubstitutionTest.java

```diff
@@ -0,0 +1,160 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.metrics;
+
+import io.questdb.ServerMain;
+import io.questdb.std.Misc;
+import io.questdb.std.str.StringSink;
+import io.questdb.test.AbstractBootstrapTest;
+import io.questdb.test.TestServerMain;
+import org.junit.Assert;
+import org.junit.Test;
+import org.postgresql.util.PSQLException;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static io.questdb.test.cutlass.pgwire.BasePGTest.assertResultSet;
+import static io.questdb.test.cutlass.pgwire.BasePGTest.printToSink;
+import static io.questdb.test.tools.TestUtils.assertMemoryLeak;
+
+@SuppressWarnings("SqlDialectInspection")
+public class QueryTracingSubstitutionTest extends AbstractBootstrapTest {
+
+    @Test
+    public void testSubstitutionOfBindVariables() throws Exception {
+        assertMemoryLeak(() -> {
+            createDummyConfiguration("pg.select.cache.enabled=true", "query.tracing.enabled=true", "query.tracing.bind.variable.substitution.enabled=true");
+            try (final ServerMain serverMain = TestServerMain.createWithManualWalRun(getServerMainArgs())) {
+                serverMain.start();
+
+                try (Connection connection = getConnection(serverMain)) {
+                    try (final PreparedStatement stmt = connection.prepareStatement("SELECT ?, ?, ?, ?, ?, ?, ?, ?;")) {
```

Comment:
a lot of the `<Type>BindVariable.toSink()` methods are not or only partially covered by tests.
would be good to include all types with NULL and non-NULL values in this test.